### DISCUSSION TITLE
Fonts Guide

### DISCUSSION THREAD

**Discussion by ayuhito** [remix-run/remix#9251] [Proposals]
A lot of other frameworks ([Astro](https://docs.astro.build/en/guides/fonts), [Next.js](https://nextjs.org/docs/app/building-your-application/optimizing/fonts), [Qwik](https://qwik.dev/docs/cookbook/fonts/#fonts), etc.) have a guide on the "blessed" practices of importing fonts. A short guide on importing Google Fonts, Fontsource (I maintain this), custom fonts and extras such as preloading would be helpful for new developers. 

I'd love to make a PR on this but I just wanted a go ahead from the Remix team before doing so. 👍

**Comment by nikolailehbrink** [comment]
Since many developers seem to be looking for guidance on this, I put together my own font implementation guide for Remix and React Router 7. It covers everything @ayuhito brought up here (thanks again for maintaining Fontsource!), except Google Fonts—that part is already well-handled in the starter template, I think.

You can check it out here, hope it helps: [The missing guide to fonts in Remix.run and React Router 7](https://www.nikolailehbr.ink/blog/fonts-remix-react-router-7)

**Comment by monolithed** [comment]
I think what's needed here isn't a guide, but font management, that layer is simply missing right now. Here's how easily a font is connected in Next.js:

```tsx
import {Inter} from 'next/font/google';

const inter = Inter({
    subsets: ['latin', 'cyrillic'],
    variable: '--font-inter',
    weight: ['400', '500', '600',  '800'],
    display: 'optional',
    fallback: ['Helvetica', 'Arial', 'sans-serif']
});
```

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.
